task/mgmt-gateway/src/mgs_handler.rs

```diff
@@ -125,7 +141,60 @@ impl SpHandler for MgsHandler {
             *to = from;
         }
 
-        Ok(SpState { serial_number })
+        Ok(SpState {
+            serial_number,
+            version: VERSION,
+        })
+    }
+
+    fn update_start(
+        &mut self,
+        _sender: SocketAddrV6,
+        _port: SpPort,
+        update: UpdateStart,
+    ) -> Result<(), ResponseError> {
+        ringbuf_entry!(Log::MgsMessage(MgsMessage::UpdateStart {
+            length: update.total_size
+        }));
+
+        if let Some(progress) = self.update_progress.as_ref() {
+            return Err(ResponseError::UpdateInProgress {
+                bytes_received: progress.bytes_written as u32,
+            });
+        }
+
+        self.update_task
+            .prep_image_update()
+            .map_err(|err| ResponseError::UpdateFailed(err as u32))?;
+
+        // We can only call `claim_update_buffer_static` once; we bail out above
+        // if `self.update_progress` is already `Some(_)`, and after we claim it
+        // here, we store that into `self.update_progress` (and never clear it).
```

Comment:
How do we abort the update process?  Resetting the SP seems a bit heavy handed.

Reply:
https://github.com/oxidecomputer/hubris/issues/721